File: day01/part1.py
def readTxtFile(filePath):
    with open(filePath, 'r') as file:
        lines = file.readlines()
    return [line.strip() for line in lines]

def countNumberOfZeros(file_path: str):
    currentPositon = 50
    numberOfZero = 0
    lines = readTxtFile(file_path)
    for i in range(len(lines)):
        direction = lines[i][0]
        magnitude = int(lines[i][1:])
        if direction == 'R':
            currentPositon += magnitude
            while currentPositon > 100:
                currentPositon = currentPositon - 100
        elif direction == 'L':
            currentPositon -= magnitude
            while currentPositon < 0:
                currentPositon = 100 + currentPositon
        if currentPositon == 100:
            currentPositon = 0
        if currentPositon == 0:
            numberOfZero += 1
    return numberOfZero



print(countNumberOfZeros('day01.txt'))

File: day01/part1.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>


int main() {
    FILE * fptr;
    fptr = fopen("day01.txt", "r");
    
    if (fptr == NULL) {
        printf("Could not open file");
        return 1;
    }

    char myString[100];
    int currentPosition = 50;
    int zeroCount = 0;

    while(fgets(myString, 100, fptr)) {
        char direction;
        int magnitude;

        if (sscanf(myString, "%c%d", &direction, &magnitude)) {
            if (direction == 'R') {
                currentPosition = (currentPosition + magnitude) % 100;
            } else if (direction == 'L') {
                currentPosition = (currentPosition - magnitude) % 100;
                if (currentPosition < 0) {
                    currentPosition += 100;
                }
            }
            if (currentPosition == 0) {
                zeroCount ++;
            }
        }
    };
    printf("Password: %d\n", zeroCount);
    fclose(fptr);
    return 0;
}